src/components/icon/icon.spec.js

```diff
@@ -419,6 +419,29 @@ describe('mdIcon service', function() {
         $scope.$digest();
       });
 
+      describe('and the URL is a data URL', function() {
+        var svgData = '<svg><g><circle r="50" cx="100" cy="100"></circle></g></svg>';
+
+        describe('and the data is base64 encoded', function() {
+          it('should return correct SVG markup', function() {
+            var data = 'data:image/svg+xml;base64,' + btoa(svgData);
```

Comment:
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  dupuyyann@gmail.com
    Domain zyapps.com has exceeded the max emails per hour (440/400 (110%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
Received: from github-smtp2-ext6.iad.github.net ([192.30.252.197]:59544 helo=github-smtp2b-ext-cp1-prd.iad.github.net)
	by serveur.ex2-p10.com with esmtps (TLSv1.2:ECDHE-RSA-AES256-GCM-SHA384:256)
	(Exim 4.86_1)
	(envelope-from <noreply@github.com>)
	id 1afmNg-000azM-Ri
	for yann.dupuy@zyapps.com; Tue, 15 Mar 2016 11:39:28 +0100
Date: Mon, 14 Mar 2016 08:52:40 -0700
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=github.com;
	s=pf2014; t=1457970760;
	bh=j3w5xN7JVhK8ux+vLn2h6jLKu8V5sb/Gdq/x3z9e0/w=;
	h=From:Reply-To:To:In-Reply-To:References:Subject:List-ID:
	 List-Archive:List-Post:List-Unsubscribe:From;
	b=g6b2Wk95iHYQY4BpySZ9xbI+i0zRD4wCELbWCqsQAsD55MnsGQw6gZxzwHQGLEIMT
	 5H8egveiqj8NLVxxnn1Jq07tQ1SBvUwqCe7ZH9JWkF0TD359WmPL6h91fQhxyRu0PT
	 si2nsxdO929XSa4yJ1G+PdRu9Wiaa1pMMFu7uV0M=
From: Thomas Burleson <notifications@github.com>
Reply-To: angular/material <reply@reply.github.com>
To: angular/material <material@noreply.github.com>
Message-ID: <angular/material/pull/7547/r56022959@github.com>
In-Reply-To: <angular/material/pull/7547@github.com>
References: <angular/material/pull/7547@github.com>
Subject: Re: [material] fix(icon): Allow using data URLs (#7547)
Mime-Version: 1.0
Content-Type: multipart/alternative;
 boundary="--==_mimepart_56e6de48c2311_33423f85513e329c223341";
 charset=UTF-8
Content-Transfer-Encoding: 7bit
Precedence: list
X-GitHub-Sender: ThomasBurleson
X-GitHub-Recipient: yanndupuy
List-ID: angular/material <material.angular.github.com>
List-Archive: https://github.com/angular/material
List-Post: <mailto:reply@reply.github.com>
List-Unsubscribe: <mailto:unsub+000df0dc47f509773754a09a7c1569ab7fc0731c721bf70d92cf0000000112fea04892a170ce0356d7af@reply.github.com>,
 <https://github.com/notifications/unsubscribe/AA3w3FjtI6ZaMgiiOgIspTWzJt5p5yQ6ks5ptYRIgaJpZM4HvaeW>
X-Auto-Response-Suppress: All
X-GitHub-Recipient-Address: yann.dupuy@zyapps.com
X-Spam-Status: No, score=-3.6
X-Spam-Score: -35
X-Spam-Bar: ---
X-Ham-Report: Spam detection software, running on the system "serveur.ex2-p10.com",
 has NOT identified this incoming email as spam.  The original
 message has been attached to this so you can view it or label
 similar future email.  If you have any questions, see
 root\@localhost for details.
 
 Content preview:  > @@ -419,6 +419,29 @@ describe('mdIcon service', function()
    { > $scope.$digest(); > }); > > + describe('and the URL is a data URL', function()
    { > + var svgData = '<svg><g><circle r="50" cx="100" cy="100"></circle></g></svg>';
    > + > + describe('and the data is base64 encoded', function() { > + it('should
    return correct SVG markup', function() { > + var data = 'data:image/svg+xml;base64,'
    + btoa(svgData); [...] 
 
 Content analysis details:   (-3.6 points, 5.0 required)
 
  pts rule name              description
 ---- ---------------------- --------------------------------------------------
 -5.0 RCVD_IN_DNSWL_HI       RBL: Sender listed at http://www.dnswl.org/, high
                             trust
                             [192.30.252.197 listed in list.dnswl.org]
 -0.0 RCVD_IN_MSPIKE_H2      RBL: Average reputation (+2)
                             [192.30.252.197 listed in wl.mailspike.net]
  0.8 DATE_IN_PAST_12_24     Date: is 12 to 24 hours before Received: date
 -0.0 SPF_PASS               SPF: sender matches SPF record
  0.7 HTML_IMAGE_ONLY_20     BODY: HTML: images with 1600-2000 bytes of words
  0.0 HTML_MESSAGE           BODY: HTML included in message
 -0.1 DKIM_VALID_AU          Message has a valid DKIM or DK signature from author's
                             domain
  0.1 DKIM_SIGNED            Message has a DKIM or DK signature, not necessarily valid
 -0.1 DKIM_VALID             Message has at least one valid DKIM or DK signature
X-Spam-Flag: NO


----==_mimepart_56e6de48c2311_33423f85513e329c223341
Content-Type: text/plain;
 charset=UTF-8
Content-Transfer-Encoding: 7bit

> @@ -419,6 +419,29 @@ describe('mdIcon service', function() {
>          $scope.$digest();
>        });
>  
> +      describe('and the URL is a data URL', function() {
> +        var svgData = '<svg><g><circle r="50" cx="100" cy="100"></circle></g></svg>';
> +
> +        describe('and the data is base64 encoded', function() {
> +          it('should return correct SVG markup', function() {
> +            var data = 'data:image/svg+xml;base64,' + btoa(svgData);

In `demoLoadSvgIconsFromUrl/script.js` you have a function `$scope.getAndroidEncoded()`, I do not see a unit test for that type of solution.

---
Reply to this email directly or view it on GitHub:
https://github.com/angular/material/pull/7547/files#r56022959
----==_mimepart_56e6de48c2311_33423f85513e329c223341
Content-Type: text/html;
 charset=UTF-8
Content-Transfer-Encoding: 7bit

<p>In <a href="https://github.com/angular/material/pull/7547#discussion_r56022959">src/components/icon/icon.spec.js</a>:</p>
<pre style='color:#555'>&gt; @@ -419,6 +419,29 @@ describe(&#39;mdIcon service&#39;, function() {
&gt;          $scope.$digest();
&gt;        });
&gt;  
&gt; +      describe(&#39;and the URL is a data URL&#39;, function() {
&gt; +        var svgData = &#39;&lt;svg&gt;&lt;g&gt;&lt;circle r=&quot;50&quot; cx=&quot;100&quot; cy=&quot;100&quot;&gt;&lt;/circle&gt;&lt;/g&gt;&lt;/svg&gt;&#39;;
&gt; +
&gt; +        describe(&#39;and the data is base64 encoded&#39;, function() {
&gt; +          it(&#39;should return correct SVG markup&#39;, function() {
&gt; +            var data = &#39;data:image/svg+xml;base64,&#39; + btoa(svgData);
</pre>
<p>In <code>demoLoadSvgIconsFromUrl/script.js</code> you have a function <code>$scope.getAndroidEncoded()</code>, I do not see a unit test for that type of solution.</p>

<p style="font-size:small;-webkit-text-size-adjust:none;color:#666;">&mdash;<br>Reply to this email directly or <a href="https://github.com/angular/material/pull/7547/files#r56022959">view it on GitHub</a>.<img alt="" height="1" src="https://github.com/notifications/beacon/AA3w3FK32Jb__lWuHjs5UHrGvZW_Lssoks5ptYRIgaJpZM4HvaeW.gif" width="1" /></p>
<div itemscope itemtype="http://schema.org/EmailMessage">
<div itemprop="action" itemscope itemtype="http://schema.org/ViewAction">
  <link itemprop="url" href="https://github.com/angular/material/pull/7547/files#r56022959"></link>
  <meta itemprop="name" content="View Pull Request"></meta>
</div>
<meta itemprop="description" content="View this Pull Request on GitHub"></meta>
</div>

----==_mimepart_56e6de48c2311_33423f85513e329c223341--